From 42bd2026d6ca9e343f60ae92e7f620087668b5ad Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Thu, 17 Aug 2023 12:25:06 +0530
Subject: [PATCH] workaround solution to allow capture/record even on storage
 error

Camera2 app is disabling all the user interaction as storage
access to data/app write permission is not allowed.

Ignoring the storage write access failure and continue the camera
app to take picture and record video.

Tracked-On: OAM-111650
Change-Id: I6ecf0b51ef758f0c03829965ca1225b2356ea3e5
Signed-off-by: shivasku82 <shiva.kumara.rudrappa@intel.com>
---
 src/com/android/camera/CameraActivity.java | 5 +++--
 src/com/android/camera/VideoModule.java    | 4 ++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index e1e80d2d9..bac0318b6 100644
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -2377,11 +2377,12 @@ public class CameraActivity extends QuickActivity
             } else {
                 mStorageHint.setText(message);
             }
-            mStorageHint.show();
+            /*mStorageHint.show();
             UsageStatistics.instance().storageWarning(storageSpace);
 
             // Disable all user interactions,
-            mCameraAppUI.setDisableAllUserInteractions(true);
+            mCameraAppUI.setDisableAllUserInteractions(true);*/
+            mCameraAppUI.setDisableAllUserInteractions(false);
         } else if (mStorageHint != null) {
             mStorageHint.cancel();
             mStorageHint = null;
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index f63c962f3..c7b69b167 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1363,9 +1363,9 @@ public class VideoModule extends CameraModule
         mActivity.updateStorageSpaceAndHint(new CameraActivity.OnStorageUpdateDoneListener() {
             @Override
             public void onStorageUpdateDone(long bytes) {
-                if (bytes <= Storage.LOW_STORAGE_THRESHOLD_BYTES) {
+/*                if (bytes <= Storage.LOW_STORAGE_THRESHOLD_BYTES) {
                     Log.w(TAG, "Storage issue, ignore the start request");
-                } else {
+                } else */{
                     if (mCameraDevice == null) {
                         Log.v(TAG, "in storage callback after camera closed");
                         return;
-- 
2.17.1

